src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -553,8 +553,15 @@ private void AttributePrefix(IEnumerable<HtmlSymbol> whitespace, IEnumerable<Htm
             }
             else
             {
+                // Output the attribute name, the equals and optional quote. Ex: foo="
+                Output(SpanKind.Markup);
+
                 // Not a "conditional" attribute, so just read the value
                 SkipToAndParseCode(sym => IsEndOfAttributeValue(quote, sym));
+
+                // Output the attribute value (will include everything inbetween the attribugte's quotes).
+                Output(SpanKind.Markup);
```

Comment:
Correct, this and the `Output(SpanKind.Markup)` for the name are important for splitting up the attribute. Typically attributes are always split up like so: `class="hello"` => `|class="|hello|"|` or `class="prefix @DateTime.Now suffix"` => `|class="|prefix| @DateTime.Now| suffix|"|`but this wasn't happening for `data-` attributes, it was grouping everything together.